src/java/org/apache/cassandra/cql3/selection/ResultSetBuilder.java

```diff
@@ -98,6 +102,27 @@ public void add(ByteBuffer v)
         inputRow.add(v);
     }
 
+    public void add(ComplexColumnData complexColumnData, Function<Iterator<Cell<?>>, ByteBuffer> serializer)
+    {
+        if (complexColumnData == null)
+        {
+            inputRow.add(null);
+            return;
+        }
+
+        long timestamp = -1L;
+        if (selectors.collectMaxTimestamps())
+        {
+            Iterator<Cell<?>> cells = complexColumnData.iterator();
+            while (cells.hasNext())
+            {
+                timestamp = Math.max(timestamp, cells.next().timestamp());
+            }
+        }
```

Comment:
We could use streams here, although we might need to worry about performance 
```suggestion
        long timestamp = -1L;
        if (selectors.collectMaxTimestamps())
        {
            timestamp = StreamSupport.stream(complexColumnData.spliterator(), false)
                                     .mapToLong(Cell::timestamp)
                                     .max()
                                     .orElse(-1L);
        }
```

Reply:
Only 1 minus line. Not very motivative.